### PR TITLE
Iteration 2 "pseudo code"

### PR DESCRIPTION
**Pull request by mcat56**
@sasloan @freeheeling @zacisaacson 

I think by doing, this is what I came up with. Let me know if you find anything!

I thought we name the modules as _stats since all the methods will be stat generating

### CODE REVIEW

**Comment by zacisaacson** [comment]
Looks Great!

**Comment by freeheeling** [comment]
mcat56 wants to merge 15 commits into turingschool-examples:master from zacisaacson:Iteration1

Is this where you want to merge?

File: lib/game_stats.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    #Average number of goals scored in a game organized in a hash with season
+    # names (e.g. 20122013) as keys and a float representing the average number of
+    #  goals in a game for that season as a key (rounded to the nearest 100th)
+    # goals_by_season = {}
+    goals_by_season = Hash.new(0)
+    @game_objs.each do |object|
+      goals_by_season[object.season] += (object.away_goals + object.home_goals)
+    end
+    goals_by_season.each do |season, goals|
+      goals_by_season[season] = goals / count_of_games_by_season[season].to_f
```
Comment (freeheeling) [inline lib/game_stats.rb]: Round

File: lib/games.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      @type = line.split(",")[2]
+      @date_time = line.split(",")[3]
+      @away_team_id = line.split(",")[4].to_i
+      @home_team_id = line.split(",")[5].to_i
+      @away_goals = line.split(",")[6].to_i 
+      @home_goals = line.split(",")[7].to_i
+      @venue = line.split(",")[8]
+      @venue_link = line.split(",")[9]
+    end
+
```
Comment (freeheeling) [inline lib/games.rb]: Remove whitespace

File: lib/game_stats.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    games_by_season = Hash.new(0)
+    @game_objs.each do |object|
+      games_by_season[object.season] += 1
+    end
+    games_by_season
+  end
+
+  def average_goals_per_game #float
+    #Average number of goals scored in a game across all seasons including both home and away goals (rounded to the nearest 100th)
+    @game_objs.inject(0) {|sum, object| sum += (object.home_goals + object.away_goals)} / @game_objs.length.to_f.round(2)
```
Comment (freeheeling) [inline lib/game_stats.rb]: Does `.round(2)` achieve the same outcome as `* 100`?

File: test/game_teams_test.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  def setup
+    @line_2 = GameTeams.new("2012030221,3,away,LOSS,OT,John Tortorella,2,8,44,8,3,0,44.8,17,7")
+  end
+
+  def test_it_exists
+    assert_instance_of GameTeams, @line_2
+  end
+
+
```
Comment (freeheeling) [inline test/game_teams_test.rb]: Remove line

File: test/games_test.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    assert_equal "5/16/13", @line_2.date_time
+    assert_equal 3, @line_2.away_team_id
+    assert_equal 6, @line_2.home_team_id
+    assert_equal 2, @line_2.away_goals
+    assert_equal 3, @line_2.home_goals
+    assert_equal "Toyota Stadium", @line_2.venue
+    assert_equal "/api/v1/venues/null", @line_2.venue_link
+  end
+
+
```
Comment (freeheeling) [inline test/games_test.rb]: Remove whitespace

File: test/stat_tracker_test.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
+require_relative 'test_helper'
+require_relative '../lib/stat_tracker'
+require_relative '../lib/games'
+require_relative '../lib/teams'
+require_relative '../lib/game_teams'
+
+
```
Comment (freeheeling) [inline test/stat_tracker_test.rb]: Remove whitespace

File: test/teams_test.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  def setup
+    @line_2 = Team.new("1,23,Atlanta United,ATL,Mercedes-Benz Stadium,/api/v1/teams/1")
+  end
+
+  def test_it_exists
+    assert_instance_of Team, @line_2
+  end
+
+
```
Comment (freeheeling) [inline test/teams_test.rb]: Delete line

**Pull request closed without merging**